### PR TITLE
Correct search_after handling

### PR DESCRIPTION
**Pull request by aleksmaus**
Correct the initial "naive" handling of search_after parameters.

### CODE REVIEW

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java [hunk 1]
```diff
@@ -74,16 +83,53 @@ protected QueryBuilder parseQuery(XContentParser parser) throws IOException {
...
-    private List<String> randomSearchAfter() {
-        if (randomBoolean()) {
-            return Collections.emptyList();
-        } else {
-            int size = randomIntBetween(1, 50);
-            List<String> arr = new ArrayList<>(size);
-            for (int i = 0; i < size; i++) {
-                arr.add(randomAlphaOfLength(randomIntBetween(1, 15)));
+    private SearchAfterBuilder randomJsonSearchFromBuilder() throws IOException {
+        int numSearchAfter = randomIntBetween(1, 10);
```
Comment (imotov) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: How about an empty list?

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java [hunk 2]
```diff
@@ -74,16 +83,53 @@ protected QueryBuilder parseQuery(XContentParser parser) throws IOException {
...
+        XContentBuilder jsonBuilder = XContentFactory.jsonBuilder();
+        jsonBuilder.startObject();
+        jsonBuilder.startArray("search_after");
+        for (int i = 0; i < numSearchAfter; i++) {
+            int branch = randomInt(9);
```
Comment (imotov) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: I think there is `LuceneTests.randomSortValue()` that generates random sort values. I wonder if we could use it here.

Comment (imotov) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: If it is not possible, I think there is a [better pattern for this](https://github.com/elastic/elasticsearch/blob/76de93c25808b51b27f5d3cedaee159cf3d74fea/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java#L588) base on `randomFrom(...)` and `Supplier<...>`

**Review by imotov — COMMENTED** [review]
Good catch! Left some suggestions.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java (hunk 1 above)
Comment (aleksmaus) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: normally SearchAfter doesn't accept the empty list, throws error:
```
    java.lang.IllegalArgumentException: Values must contains at least one value.
```
The original searchafter test doesn't use empty list as well
https://github.com/elastic/elasticsearch/blob/df83eb983c5f5cdc50c3d2c3d080b4c56bc571f4/server/src/test/java/org/elasticsearch/search/searchafter/SearchAfterBuilderTests.java#L101

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/RestEqlSearchAction.java [hunk 3]
```diff
@@ -29,10 +29,9 @@
 
 public class RestEqlSearchAction extends BaseRestHandler {
     public RestEqlSearchAction(RestController controller) {
-        // Not sure yet if we will always have index in the path or not
-        // TODO: finalize the endpoints
-        controller.registerHandler(GET, "/{index}/_eql/search", this);
-        controller.registerHandler(POST, "/{index}/_eql/search", this);
+        final String path = "/{index}/_eql/search";
```
[the remark this answers is not in the document]
Comment (aleksmaus) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/RestEqlSearchAction.java]: done

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java (hunk 2 above)
Comment (aleksmaus) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: updated to a better pattern

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java (hunk 1 above)
Comment (imotov) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: How about returning setting search after values to null sometimes then?

Comment (aleksmaus) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: Already handled there as well
```
    java.lang.NullPointerException: Values cannot be null.
        at __randomizedtesting.SeedInfo.seed([1C63F70E75AEE77B:48BB964652FCDAE7]:0)
        at org.elasticsearch.search.searchafter.SearchAfterBuilder.setSortValues(SearchAfterBuilder.java:78)
```

Comment (aleksmaus) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: discussed over slack, updated implementation

**Review by imotov — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**